rir/src/interpreter/interp.cpp

```diff
@@ -230,6 +230,36 @@ RIR_INLINE void __listAppend(SEXP* front, SEXP* last, SEXP value, SEXP name) {
     *last = app;
 }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-align"
+
+SEXP createEnvironment(const std::vector<SEXP>* args, const SEXP parent,
+                       const Opcode* pc, Context* ctx, R_bcstack_t* localsBase,
+                       SEXP stub) {
+    SEXP arglist = R_NilValue;
+    auto names = (Immediate*)pc;
+    for (size_t i = 0; i < args->size(); ++i) {
+        SEXP val = args->at(i);
+        SEXP name = cp_pool_at(ctx, names[i]);
+        arglist = CONS_NR(val, arglist);
+        SET_TAG(arglist, name);
+        SET_MISSING(arglist, val == R_MissingArg ? 2 : 0);
+    }
+    SEXP environment = Rf_NewEnvironment(R_NilValue, arglist, parent);
+
+    // TODO: Find a proper way of going through the function stack and
+    // replacing the stub with the new environment
+    Locals local(localsBase, 5);
+    for (auto i = 0; i < 5; i++) {
```

Comment:
why 5? I think you have to go through the whole frame, ie. from current stack top until localsBase. You could have arbitrary many values on the stack.